### PR TITLE
Fluentd Log Centralization Part 2

### PR DESCRIPTION
**Pull request by rkooo567**
# Related issue 
https://github.com/ucbrise/clipper/issues/625

# Summary
- Cleaned up the previous fluentd code
- Modified a Fluentd Dockerfile to support sqlite3 output plugin
- Modified fluentd.conf to support sqlite3 output plugin
- Now our centralized logs are streamed into sqlite3 db (it is inside fluentd container) as well as stdout

# Note
Currently, it does not support 3 things yet. It will be handled in the future PR.
- Persist logging db file in the local filesystem. Once containers shut down, all the logs are gone.
- It cannot customize fluentd.conf file programatically. Users should modify fluentd.conf mounted on the local filesystem. 
- It cannot forward logs to other logging centralization tools yet.

### CODE REVIEW

**Comment by rkooo567** [comment]
@simon-mo 

```
Problem starting Clipper: 404 Client Error: Not Found ("pull access denied for clipper/fluentd, repository does not exist or may require 'docker login'") 
```

I think we should create a repo in dockerhub?

File: clipper_admin/clipper_admin/docker/logging/fluentd.py [hunk 1]
```diff
@@ -1,41 +1,45 @@
...
+import logging
 
 from clipper_admin.container_manager import (
     CLIPPER_INTERNAL_FLUENTD_PORT, CLIPPER_FLUENTD_CONFIG_LABEL,
     CLIPPER_DOCKER_PORT_LABELS
 )
 
 
-FLUENTD_VERSION = 'v1.3-debian-1'  # TODO needs to be update to receive env variable like prometheus
+FLUENTD_REGISTRY            = 'clipper/fluentd'
```
Comment (simon-mo) [inline clipper_admin/clipper_admin/docker/logging/fluentd.py]: Please follow the `__registry__` convention as in https://github.com/ucbrise/clipper/blob/df96eb4924809fedc5a70ef1bc080129c2bf962c/clipper_admin/clipper_admin/deployers/python.py#L199

**Review by simon-mo — CHANGES_REQUESTED** [review]

File: dockerfiles/FluentdLoggingContainerDockerfile [hunk 2]
```diff
@@ -1 +1,28 @@
-# Currently, we are using a default Fluentd image from Dockerhub. Needs to be customized by Clipper
\ No newline at end of file
+FROM fluent/fluentd:v1.5-debian-1
```
Comment (simon-mo) [inline dockerfiles/FluentdLoggingContainerDockerfile:1]: Also, the docker container need to be built. See something like 
https://github.com/ucbrise/clipper/blob/df96eb4924809fedc5a70ef1bc080129c2bf962c/bin/shipyard/clipper_docker.cfg.py#L93

**Comment by simon-mo** [comment]
The two comments above might resolve the container not found issue

**Comment by rkooo567** [comment]
@simon-mo Will check it out tomorrow! Thanks for the review!

File: clipper_admin/clipper_admin/docker/logging/fluentd.py (hunk 1 above)
Comment (rkooo567) [reply clipper_admin/clipper_admin/docker/logging/fluentd.py]: Fixed!

File: dockerfiles/FluentdLoggingContainerDockerfile (hunk 2 above)
Comment (rkooo567) [reply dockerfiles/FluentdLoggingContainerDockerfile:1]: Not sure if my fix will resolve this part. Let's see if our CI can pass

**Comment by rkooo567** [comment]
@simon-mo @withsmilo @RehanSD It passes the CI. Please review it!

**Comment by rkooo567** [comment]
@simon-mo @withsmilo @RehanSD  Please review it whenever you have time!

File: clipper_admin/clipper_admin/docker/logging/fluentd.py [hunk 3]
```diff
@@ -49,12 +53,11 @@ def get_type():
         return "Fluentd"
 
     def _run_fluentd_image(self, docker_client, fluentd_labels, fluend_port, fluentd_conf_path, extra_container_kwargs):
-        fluentd_cmd = []  # No cmd is required.
+        fluentd_cmd  = []  # No cmd is required.
```
Comment (withsmilo) [inline clipper_admin/clipper_admin/docker/logging/fluentd.py:56]: Please remove an added tiny indentation.

**Comment by simon-mo** [comment]
@rkooo567 thx for the PR. I'll test this PR and benchmark ingestion speed.

**Comment by rkooo567** [comment]
@simon-mo Sounds good. @withsmilo I will handle all the change requests at once when Simon also have some comments later.

**Pull request closed without merging**